rakefile

```diff
@@ -1,13 +1,85 @@
 require 'rototiller'
 require 'rspec/core/rake_task'
+require 'net/http'
+require 'timeout'
+require 'json'
 
 task :default => :performance
 
-desc 'Run Performance setup'
-rototiller_task :performance do |t|
+desc 'Provision systems using ABS (or use already set ABS_RESOURCE_HOSTS) and then execute beaker performance setup and tests'
+rototiller_task :performance do
+  Rake::Task["performance_provision_with_abs"].execute unless ENV['ABS_RESOURCE_HOSTS']
+  Rake::Task["performance_without_provision"].execute
+end
+
+desc 'Provision systems for the performance task with ABS'
+rototiller_task :performance_provision_with_abs do
+
+  ENV['ABS_JOB_ID'] = rand(100000..999999) unless ['ABS_JOB_ID']
+  uri = URI('https://cinext-abs-test.delivery.puppetlabs.net/api/v2/request')
+  req = Net::HTTP::Post.new(uri, 'Content-Type' => 'application/json', )
+  req.body = {'resources': {'centos-7-x86-64-west': 2},'job': { 'id': "#{ENV['ABS_JOB_ID']}", 'tags': {'user': 'jenkins','jenkins_build_url': 'https://jenkins/job/local-perf-test'}}}.to_json
+  http = Net::HTTP.new(uri.host, uri.port)
+  http.use_ssl = true
+
+  puts "sending request body: #{req.body}"
+  puts "to uri: #{uri}"
+  res = nil
+
+  # Wait up to 5 minutes for a response from ABS with provisioned hosts
```

Comment:
this could probably be placed in a method, outside the task.  as-is, it's hard for me to follow what the task is doing exactly.